bokeh/model.py

```diff
@@ -739,9 +723,22 @@ def __exit__(self, type, value, traceback):
         for model in self._to_remove_after:
             model.document.remove_root(model)
 
-    def __enter__(self):
+    def __enter__(self, tried_already=False):
         for model in self._to_remove_after:
-            self._doc.add_root(model)
+            try:
+                self._doc.add_root(model)
+            except:
```

Comment:
I also still think it's worth considering just not handling this case (i.e. let the exception escape, or convert to a more meaningful exception to re-raise). 

In a broad sense, if the intent is to make something faster then having a fallback to something slow that happens silently without warning may not be advised. However, in this case I aslo really think now that `figure`, etc. don't automatically add themselves to an implicit current document on initialization, that this should really just not happen with normal usage, and if it does happen, then maybe indicates a usage error of some sort.